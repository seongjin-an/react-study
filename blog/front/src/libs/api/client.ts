import axios from "axios";

const client = axios.create()
/*
client.defaults.baseURL = 'url'
client.defaults.headers.common['Authorization'] = 'B'
axios.interceptors.response.use(
    response => {
        return response
    },
    error => {
        return Promise.reject(error)
    }
)
*/
export default client