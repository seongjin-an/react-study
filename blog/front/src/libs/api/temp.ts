import axios from "axios";
import {TempPost} from "../../types/TempPost";
import {useQuery} from "react-query";

const getPostById = async (id: number): Promise<TempPost> => {
    const { data } = await axios.get(
        `https://jsonplaceholder.typicode.com/posts/${id}`,
    );
    return data;
};

export const usePost = (postId: number) => {
    return useQuery(['post', postId], () => getPostById(postId), {
        enabled: !!postId,
    });
};

const getPosts = async (): Promise<Array<TempPost>> => {
    const { data } = await axios.get(
        'https://jsonplaceholder.typicode.com/posts',
    );
    return data;
};

export const usePosts = () => {
    return useQuery('posts', getPosts);
};