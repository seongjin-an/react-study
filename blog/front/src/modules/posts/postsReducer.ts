import {createReducer} from "@reduxjs/toolkit";
import {LIST_POSTS_FAILURE, LIST_POSTS_SUCCESS} from "./postsType";
export interface IUser{
    username: string
    _id: string
}
export interface IPost{
    body: string
    publishedDate: string
    tags: string[]
    title: string
    user: IUser
    __v: number
    _id: string
}
interface IPostsState {
    posts: IPost[] | null
    error:null,
    lastPage: number
}
const initialState = {
    posts: null,
    error: null,
    lastPage: 1
} as IPostsState
const posts = createReducer(initialState,{
    [LIST_POSTS_SUCCESS]: (state: IPostsState, action) => {
        return{
            ...state,
            posts: action.payload,
            lastPage: parseInt(action.meta.headers["last-page"], 10)
        }
    },
    [LIST_POSTS_FAILURE]: (state: IPostsState, action) => {
        return{
            ...state,
            error: action.payload
        }
    }
})
export default posts