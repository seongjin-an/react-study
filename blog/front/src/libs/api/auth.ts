import client from './client'

export const login = ({username, password}:{username:string, password:string}) => {
    return client.post('/api/auth/login', {username, password})
}

export const register = ({username, password}:{username:string, password:string}) => {
    return client.post('/api/auth/register', {username, password})
}
export const check = () => client.get('/api/auth/check')

export const logout = () => client.post('/api/auth/logout')