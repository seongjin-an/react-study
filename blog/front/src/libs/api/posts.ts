import qs from 'qs'
import client from "./client";
import {IWriteState} from "../../modules/write/writeReducer";
import {IPosts} from "../../modules/posts/postsType";

export const writePost = ({title, body, tags}:IWriteState) => client.post('/api/posts', {title, body, tags})

export const readPost = (id:string) => client.get(`/api/posts/${id}`)

export const listPosts = ({ page, username, tag }: IPosts) => {
    const querystring = qs.stringify({ page, username, tag })
    return client.get(`/api/posts?${querystring}`)
}