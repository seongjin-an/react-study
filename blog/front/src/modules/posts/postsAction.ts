import {createAction} from "@reduxjs/toolkit";
import {IPosts, LIST_POSTS_FAILURE, LIST_POSTS_REQUEST, LIST_POSTS_SUCCESS} from "./postsType";
import {ParsedQs} from "qs";
import {AxiosResponse} from "axios";
import {IPost} from "./postsReducer";
export interface IPostsSuccess{
    list: IPost[]
    meta: AxiosResponse
}
export const listPostsAction = createAction<IPosts>(LIST_POSTS_REQUEST)
export const listPostsSuccess = createAction<IPostsSuccess>(LIST_POSTS_SUCCESS)
export const listPostsFailure = createAction<AxiosResponse>(LIST_POSTS_FAILURE)