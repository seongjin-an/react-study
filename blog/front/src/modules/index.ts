import {combineReducers, Reducer} from "redux";
import {configureStore} from "@reduxjs/toolkit";
import createSagaMiddleware from "@redux-saga/core";
import {all, fork} from "@redux-saga/core/effects";
import auth from "./auth/authReducer";
import loading from './loading/loadingReducer'
import user from './user/userReducer'
import write from './write/writeReducer'
import post from "./post/postReducer";

import authSaga from "./auth/authSaga";
import userSaga from "./user/userSaga";
import writeSaga from "./write/writeSaga";
import postSaga from "./post/postSaga";

const rootReducer:Reducer = combineReducers({
    auth,
    loading,
    user,
    write,
    post
})
export const rootSaga = function*(){
    yield all([
        fork(authSaga),
        fork(userSaga),
        fork(writeSaga),
        fork(postSaga)
    ])
}
export type RootState = ReturnType<typeof rootReducer>
export const createStore = () => {
    const sagaMiddleware = createSagaMiddleware();
    const store = configureStore({
        reducer: rootReducer,
        middleware: [sagaMiddleware],
        devTools: process.env.NODE_ENV !== "production",
    })
    sagaMiddleware.run(rootSaga);
    return store
}

export default rootReducer