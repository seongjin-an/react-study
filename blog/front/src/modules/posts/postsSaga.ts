import createRequestSaga from "../../libs/createRequestSaga";
import {LIST_POSTS, LIST_POSTS_REQUEST} from "./postsType";
import {listPosts} from "../../libs/api/posts";
import {all, fork, takeLatest} from "@redux-saga/core/effects";

const fetchListPosts = createRequestSaga(LIST_POSTS, listPosts)
const watchListPosts = function*(){
    yield takeLatest(LIST_POSTS_REQUEST, fetchListPosts)
}
const postsSaga = function*(){
    yield all([
        fork(watchListPosts)
    ])
}
export default postsSaga