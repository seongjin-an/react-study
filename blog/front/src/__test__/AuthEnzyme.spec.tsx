import {configure, mount, shallow} from 'enzyme'
import React from "react";
import {LoginPage} from "../pages/login";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {BrowserRouter} from "react-router-dom";
import {Store} from "@reduxjs/toolkit";
import {createStore} from "../modules";
import {Provider} from "react-redux";
import toJson from "enzyme-to-json";
import "@testing-library/jest-dom/extend-expect";
import {listPosts} from "../libs/api/posts";

let store: Store
const setup = (WrappedComponent: React.ComponentProps<any>, props={}) => {
    const wrapper = mount(
        <Provider store={store}>
            <BrowserRouter>
                <WrappedComponent {...props} />
            </BrowserRouter>
        </Provider>
    )
    return {
        wrapper
    }
}

describe('auth component', () => {

    beforeEach(async () => {
        store = await createStore()
        configure({ adapter: new Adapter() });
    })
    it('should match snapshot', async () => {
        const {wrapper} = await setup(LoginPage)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('should render', async () => {
        const {wrapper} = await setup(LoginPage)
        const header=wrapper.find('.logo-area')
        expect(header.text()).toBe('REACTERS')
        const loginButton = wrapper.find('AuthForm').find('CommonButton')
        expect(loginButton.text()).toBe('로그인')
        expect(loginButton.text()).not.toBe('회원가입')
    })
    it('should login', async () => {
        const {wrapper} = await setup(LoginPage)
        wrapper.find('input[name="username"]')
            .simulate('change', {target: {value: 'imsi11'}})
        wrapper.find('input[name="password"]')
            .simulate('change', {target: {value: '123411'}})
        wrapper.find('button').simulate('click')
    })
})
