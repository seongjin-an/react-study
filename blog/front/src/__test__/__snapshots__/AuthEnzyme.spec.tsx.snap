// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <LoginPage>
        <AuthTemplate>
          <styled.div>
            <div
              className="sc-bdfBQB binGTg"
            >
              <styled.div>
                <div
                  className="sc-gsTEea dahxES"
                >
                  <div
                    className="logo-area"
                  >
                    <Link
                      to="/"
                    >
                      <LinkAnchor
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          REACTERS
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                  <AuthArea
                    type={0}
                  >
                    <styled.div>
                      <div
                        className="sc-gKseQn gNkDtN"
                      >
                        <h3>
                          로그인
                        </h3>
                        <AuthForm
                          error=""
                          onChange={[Function]}
                          onSubmit={[Function]}
                          type={0}
                        >
                          <form>
                            <AuthInput
                              autoComplete="username"
                              name="username"
                              onChange={[Function]}
                              placeholder="아이디"
                              type="text"
                            >
                              <styled.input
                                autoComplete="username"
                                name="username"
                                onChange={[Function]}
                                placeholder="아이디"
                                type="text"
                              >
                                <input
                                  autoComplete="username"
                                  className="sc-dlfnuX duNDNx"
                                  name="username"
                                  onChange={[Function]}
                                  placeholder="아이디"
                                  type="text"
                                />
                              </styled.input>
                            </AuthInput>
                            <AuthInput
                              autoComplete="new-password"
                              name="password"
                              onChange={[Function]}
                              placeholder="비밀번호"
                              type="password"
                            >
                              <styled.input
                                autoComplete="new-password"
                                name="password"
                                onChange={[Function]}
                                placeholder="비밀번호"
                                type="password"
                              >
                                <input
                                  autoComplete="new-password"
                                  className="sc-dlfnuX duNDNx"
                                  name="password"
                                  onChange={[Function]}
                                  placeholder="비밀번호"
                                  type="password"
                                />
                              </styled.input>
                            </AuthInput>
                            <CommonButton
                              cyan={true}
                              fullWidth={true}
                              onClick={[Function]}
                              style={
                                Object {
                                  "marginTop": "1rem",
                                }
                              }
                            >
                              <styled.button
                                cyan={true}
                                fullWidth={true}
                                onClick={[Function]}
                                style={
                                  Object {
                                    "marginTop": "1rem",
                                  }
                                }
                              >
                                <button
                                  className="sc-eCstlR eeLcBN"
                                  onClick={[Function]}
                                  style={
                                    Object {
                                      "marginTop": "1rem",
                                    }
                                  }
                                >
                                  로그인
                                </button>
                              </styled.button>
                            </CommonButton>
                          </form>
                          <AuthFooter>
                            <styled.div>
                              <div
                                className="sc-hKgJUU jwSaWL"
                              >
                                <Link
                                  to="/register"
                                >
                                  <LinkAnchor
                                    href="/register"
                                    navigate={[Function]}
                                  >
                                    <a
                                      href="/register"
                                      onClick={[Function]}
                                    >
                                      회원가입
                                    </a>
                                  </LinkAnchor>
                                </Link>
                              </div>
                            </styled.div>
                          </AuthFooter>
                        </AuthForm>
                      </div>
                    </styled.div>
                  </AuthArea>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </AuthTemplate>
      </LoginPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
